generate_data.py

```diff
@@ -0,0 +1,170 @@
+import argparse
+import ast
+import copy
+import shutil
+import sys
+import os
+import yaml
+
+import numpy as np
+from tqdm import tqdm
+
+from environments import environment_factory
+
+
+def generate_and_save(root_path, environment, n_samples, n_frames, delta_time, img_size,
+                      radius_bound, noise_level, color, start_seed, train=True):
+    path = os.path.join(root_path, 'train' if train else 'test')
+    if not os.path.exists(path):
+        os.makedirs(path)
+    for i in tqdm(range(n_samples)):
+        rolls = environment.sample_random_rollouts(
+            number_of_frames=n_frames,
+            delta_time=delta_time,
+            number_of_rollouts=1,
+            img_size=img_size,
+            noise_level=noise_level,
+            radius_bound=radius_bound,
+            color=color,
+            seed=i + start_seed
+        )[0]
+        filename = "{0:05d}".format(i)
+        np.savez(os.path.join(path, filename), rolls)
+    return path
+
+
+def _read_params(params_file):
+    with open(params_file, 'r') as f:
+        params = yaml.load(f, Loader=yaml.FullLoader)
+    return params
+
+
+def _prepare_out_params(params, train_path, test_path):
+    offline_params = copy.deepcopy(params)
+    offline_params.pop('img_size', None)
+    offline_params.pop('radius_bound', None)
+    offline_params.pop('num_train_samples', None)
+    offline_params.pop('num_test_samples', None)
+    offline_params['train_data'] = train_path
+    offline_params['test_data'] = test_path
+    return offline_params
+
+
+def _overwrite_params_with_cmd_arguments(params, args):
+    # This function overwrites parameters in the given dictionary
+    # with the correspondent command line arguments.
+    if args.name is not None:
+        params['experiment_id'] = args.name[0]
+    if args.ntrain is not None:
+        params['dataset']['num_train_samples'] = args.ntrain[0]
+    if args.ntest is not None:
+        params['dataset']['num_test_samples'] = args.ntest[0]
+    if args.env is not None:
+        env_params = _read_params(DEFAULT_ENVIRONMENTS_PATH + args.env[0] + '.yaml')
+        params['environment'] = env_params['environment']
+    if args.env_spec is not None:
+        for spec in args.env_spec:
+            key, value = spec.split(':')
+            params['environment'][key] = ast.literal_eval(value)
+
+
+if __name__ == '__main__':
+    DATASETS_ROOT = os.path.join(os.path.dirname(__file__), 'datasets')
```

Comment:
Could we make this be a parameter in the yaml file?

In my case i need to load the datasets in a different location than "datasets". This change also applies to offline dataset loader called in the train.py file. I might open a new PR for that.

Reply:
Can I make it a command line argument? Because it would be weird to have it the `default_online.yaml`

Reply:
oh that would be perfect.